common/src/chunk_ray_casting.rs

```diff
@@ -0,0 +1,318 @@
+            // to this voxel has grid coordinates from (1, 1, 1) to (2, 2, 2)
+            ctx.set_voxel([1, 1, 1], Material::Dirt);
+
+            ctx
+        }
+
+        fn set_voxel(&mut self, coords: [u8; 3], material: Material) {
+            debug_assert!(coords[0] < self.layout.dimension());
+            debug_assert!(coords[1] < self.layout.dimension());
+            debug_assert!(coords[2] < self.layout.dimension());
+            self.voxel_data.data_mut(self.layout.dimension())
+                [Coords(coords).to_index(self.layout.dimension())] = material;
+        }
+    }
+
+    /// Helper method to create a `ChunkSphereCastContext` that can be used
+    /// in a closure to call sphere casting methods.
+    fn cast_with_test_ray(
+        ctx: &TestRayCastContext,
+        ray_start_grid_coords: [f32; 3],
+        ray_end_grid_coords: [f32; 3],
+        wrapped_fn: impl FnOnce(&Ray, f32),
+    ) {
+        let ray_start = math::lorentz_normalize(&na::Vector4::new(
+            ray_start_grid_coords[0] / ctx.layout.dual_to_grid_factor(),
+            ray_start_grid_coords[1] / ctx.layout.dual_to_grid_factor(),
+            ray_start_grid_coords[2] / ctx.layout.dual_to_grid_factor(),
+            1.0,
+        ));
+
+        let ray_end = math::lorentz_normalize(&na::Vector4::new(
+            ray_end_grid_coords[0] / ctx.layout.dual_to_grid_factor(),
+            ray_end_grid_coords[1] / ctx.layout.dual_to_grid_factor(),
+            ray_end_grid_coords[2] / ctx.layout.dual_to_grid_factor(),
+            1.0,
+        ));
+
+        let ray = Ray::new(
+            ray_start,
+            math::lorentz_normalize(
+                &((ray_end - ray_start)
+                    + ray_start * math::mip(&ray_start, &(ray_end - ray_start))),
+            ),
+        );
+
+        let tanh_distance = (-math::mip(&ray_start, &ray_end)).acosh();
+
+        wrapped_fn(&ray, tanh_distance)
+    }
+
+    fn chunk_ray_cast_wrapper(
+        ctx: &TestRayCastContext,
+        ray: &Ray,
+        tanh_distance: f32,
+    ) -> Option<ChunkCastHit> {
+        chunk_ray_cast(&ctx.voxel_data, &ctx.layout, ray, tanh_distance)
+    }
+
+    fn test_face_collision(
+        ctx: &TestRayCastContext,
+        ray: &Ray,
+        tanh_distance: f32,
+        expected_face_axis: CoordAxis,
+        expected_face_direction: CoordDirection,
+    ) {
+        let hit = chunk_ray_cast_wrapper(ctx, ray, tanh_distance);
+        let hit = hit.expect("collision expected");
+        assert_eq!(hit.voxel_coords, Coords([1, 1, 1]));
+        assert_eq!(hit.face_axis, expected_face_axis);
+        assert_eq!(hit.face_direction, expected_face_direction);
+        // sanity_check_normal(ray, &hit.unwrap()); TODO: Check other results
+    }
+
+    /// Tests that a suitable collision is found when approaching a single voxel from various angles and that
+    /// no collision is found in paths that don't reach that voxel.
+    #[test]
+    fn chunk_sphere_cast_examples() {
```

Comment:
Good catch. I've also noticed a few issues related to your comment which I'll need to fix (mentioning in case you want to double-check my fixes):
- There is also a mention of spheres in this file in the docs to `cast_with_test_ray`
- The docs mention a `ChunkSphereCastContext` even though that has been refactored away.